### PR TITLE
refactor: use of lock file in ModuleGraph2

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes https://github.com/denoland/deno/issues/8065

### CODE REVIEW

File: cli/module_graph2.rs [hunk 1]
```diff
@@ -1393,13 +1397,8 @@ impl GraphBuilder2 {
...
   /// the expected lockfile, an error will be logged and the process will exit.
-  ///
-  /// TODO(@kitsonk) this should really be owned by the graph, but currently
-  /// the lockfile is behind a mutex in program_state, which makes it really
-  /// hard to not pass around as a reference, which if the Graph owned it, it
-  /// would need lifetime parameters and lifetime parameters are 😭
-  pub fn get_graph(self, maybe_lockfile: &Option<Mutex<Lockfile>>) -> Graph2 {
-    self.graph.lock(maybe_lockfile);
+  pub fn get_graph(self) -> Graph2 {
+    self.graph.lock(self.maybe_lockfile.as_ref());
```
Comment (kitsonk) [inline cli/module_graph2.rs:1401]: Why not pass it to the graph itself in `::new()` like the `handler`?  Then `self.graph.lock` and just `self.maybe_lockfile` itself.

**Review by kitsonk — COMMENTED** [review]
Just 1 thought, everything else LGTM.

File: cli/module_graph2.rs (hunk 1 above)
Comment (bartlomieju) [reply cli/module_graph2.rs]: Done

**Review by kitsonk — APPROVED** [review]
LGTM

**Pull request merged**